packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -869,7 +870,7 @@ class _AppBarState extends State<AppBar> {
           tooltip: MaterialLocalizations.of(context).openAppDrawerTooltip,
         );
       } else {
-        if (!hasEndDrawer && canPop)
+        if (canPop && !isFirstRoute)
```

Comment:
This is not right, even if it is the first route it can still pop if willHandlePopInternally is true

Reply:
Wait I was actually confused a bit, this seems like which design decision. If a page has local entry, for example, open drawer, it should probably not display the back button. We need to verify with native behavior.

Reply:
@chunhtai Hey, you are right that I made a mistake.

I reviewed #63272 again. This is a completely wrong change. The original issue is just a minor display optimization, but it introduced some bugs, so I decided to roll it back.
In addition, our current API seems to not easily fix #13601, so I am going to reopen it.

Reply:
if you added !isFirstRoute does it fix the  https://github.com/flutter/flutter/issues/13601?

Reply:
But it will break other test,
The dragable bottom sheet need show the back button.

Reply:
I remember there is way to get whether the drawer is open or close, and we can decide whether we want to show it or not based on that, but it is quite ugly.
if (canPop && !(isFirstRoute && drawerOpen) )

Reply:
Drag the bottom sheet and trigger the back button then open the end drawer.

Reply:
lol, I think we may need to refactor this a bit inorder to fix it